### PR TITLE
[*] FO : return a list of unavailable products

### PR DESCRIPTION
**Pull request by cedricfontaine**

### CODE REVIEW

**Comment by kpodemski** [comment]
:+1: 

but you should make it PSR-2 compatible

**Comment by jeromenadaud** [comment]
Hi @cedricfontaine 

Thank you for your contribution. Can you please use PSR-2 norm on your code ?
Best regards.

**Comment by cedricfontaine** [comment]
Ok I think I updated it.

File: controllers/front/OrderController.php [hunk 1]
```diff
@@ -55,9 +55,10 @@ public function init()
         }
 
         // If some products have disappear
-        if (is_array($product)) {
+        if (is_array($products)) {
             $this->step = 0;
-            $this->errors[] = sprintf(Tools::displayError('An item (%1s) in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.'), $product['name']);
+            foreach ($products as $product)
```
Comment (jeromenadaud) [inline controllers/front/OrderController.php]: Hi,
Missing { and }

Best regards.

File: controllers/front/OrderOpcController.php [hunk 2]
```diff
@@ -575,7 +575,10 @@ protected function _getPaymentMethods()
 
         /* If some products have disappear */
         if (is_array($product = $this->context->cart->checkQuantities(true))) {
-            return '<p class="warning">'.sprintf(Tools::displayError('An item (%s) in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.'), $product['name']).'</p>';
+            $errors = '';
+            foreach ($products as $product)
```
Comment (jeromenadaud) [inline controllers/front/OrderOpcController.php]: Hi,
Missing { and }

Best regards.

**Comment by cedricfontaine** [comment]
Ok it should be compliant.

File: controllers/front/OrderController.php [hunk 3]
```diff
@@ -55,9 +55,11 @@ public function init()
         }
 
         // If some products have disappear
-        if (is_array($product)) {
+        if (is_array($products)) {
```
Comment (maximebiloe) [inline controllers/front/OrderController.php]: `$products` is not defined ?

File: classes/Cart.php [hunk 4]
```diff
@@ -3256,17 +3256,19 @@ public function checkQuantities($return_product = false)
...
-        foreach ($this->getProducts() as $product) {
-            if (!$this->allow_seperated_package && !$product['allow_oosp'] && StockAvailable::dependsOnStock($product['id_product']) &&
-                $product['advanced_stock_management'] && (bool)Context::getContext()->customer->isLogged() && ($delivery = $this->getDeliveryOption()) && !empty($delivery)) {
-                $product['stock_quantity'] = StockManager::getStockByCarrier((int)$product['id_product'], (int)$product['id_product_attribute'], $delivery);
-            }
-            if (!$product['active'] || !$product['available_for_order']
-                || (!$product['allow_oosp'] && $product['stock_quantity'] < $product['cart_quantity'])) {
-                return $return_product ? $product : false;
-            }
+    $products = array();
```
Comment (maximebiloe) [inline classes/Cart.php]: It's not PSR-2, please correct your indentation

**Comment by maximebiloe** [comment]
Hi,

Thanks for contributing.
Your PR has a lot of commits which do not respect our commit [norm](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message). Could you change your commits messages, please  ?

Regards

**Comment by Shudrum** [comment]
Sorry, but between the commits norms, PSR, and the time passed. I close this PR.

Do not hesitate to resend one fixed if you wish.

Thank you.

**Pull request closed without merging**